diff --git a/app/Madeira/Madeira.entitlements b/app/Madeira/Madeira.entitlements
index 3c7c59f..67425f3 100644
--- a/app/Madeira/Madeira.entitlements
+++ b/app/Madeira/Madeira.entitlements
@@ -4,6 +4,8 @@
 <dict>
 	<key>com.apple.developer.kernel.increased-memory-limit</key>
 	<true/>
+	<key>com.apple.developer.kernel.extended-virtual-addressing</key>
+	<true/>
 	<key>com.apple.security.cs.allow-jit</key>
 	<true/>
 	<key>get-task-allow</key>
